### ISSUE TITLE
Rainlab.Page Url is wrong when using categories

### ISSUE THREAD

**Issue by artistro08** [OFFLINE-GmbH/oc-mall-plugin#344]
Hey man, so the category page url is wrong when navigating to a product. The generated urls picks up the hash that's generated for the current product. 

Example: 

![image](https://user-images.githubusercontent.com/6427230/66789092-b67f6300-eeaf-11e9-8899-301ac008f0a5.png)

![image](https://user-images.githubusercontent.com/6427230/66789123-d31b9b00-eeaf-11e9-8b95-119fe0d83bfa.png)

**Comment by tobias-kuendig** [comment]
You can clesr the cache to regeneate the category tree. Is it still wrong after that? Did you add the "all mall categories" entry to your menu?

**Comment by artistro08** [comment]
I just cleared the cache via artisan and it's still showing the wrong link. I did have the "All Mall Categories" entry in the menu before, but I chose the single ones because it make more sense with my design.

**Comment by tobias-kuendig** [comment]
Do you have the correct page selected as your category page in the backend settings? It should not take your product detail page when it links  to a category:

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/classes/traits/category/MenuItems.php#L126

**Comment by artistro08** [comment]
Yeah I do actually. Which is weird. I saved once more just to make sure and cleared the cache.

**Comment by tobias-kuendig** [comment]
Can you clear the cache and then check what url is being generated on this line:
https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/classes/traits/category/MenuItems.php#L133

**Comment by artistro08** [comment]
i'm not sure what you mean, but i did var_dump the item variable, and it returned this url under the product page: https://iowapopcorncompany.com/shop/product/cheddar-cheese/bRp2VpWr

this is after I cleared the cache. if needed, i can give you access to the installation

**Comment by tobias-kuendig** [comment]
Just to be sure, on the line linked above, please add this and post the output:

```
dd($pageUrl, $entryUrl);
```

**Comment by artistro08** [comment]
I did that, Loaded the page above and nothing came out. 

![image](https://user-images.githubusercontent.com/6427230/66855692-5dfca400-ef49-11e9-9571-98d08caa28d9.png)


Even Flushed the cache and checked XHR

**Comment by tobias-kuendig** [comment]
Did you by chance enable the `Replace this item with its generated children` when you added the category to your menu? Because this would insert the Products of this Category to the menu, not the Category itself.

If you disable this option you should get the Category tree, no products.

**Comment by artistro08** [comment]
I did, but that's exactly what I want. I just don't want the trailing hash that's generated when I go to the product page. I just want it to link to the product on the product page

![image](https://user-images.githubusercontent.com/6427230/66919638-51c72400-efe7-11e9-8916-7ec467e02ea8.png)

Don't want this

**Comment by tobias-kuendig** [comment]
When you have a product with variants, there is no product page. You can only link to a avariant. When you remove the id manually, you still will be redirected to the first variant of this product.

**Comment by artistro08** [comment]
I get that part. I'm just having a hard time explaining this i guess. 

When I go to the home page, the menu acts accordingly. eg:

`product => first product variant`

But when I go to the product page it does:

`product => first product variant + variant hash of current product`

You can see what I mean by heading over to these pages:

https://iowapopcorncompany.com/ = Menu Works normal
https://iowapopcorncompany.com/shop/product/cheddar-cheese/ZJ9KBDb1 = adds the `ZJ9KBDb1` to each url in the category menu.

**Comment by artistro08** [comment]
Okay awesome! Sorry for the confusion!

**Comment by artistro08** [comment]
Can confirm! Works beautifully
